### ISSUE TITLE
Seneca/Express unresponsive after first request 

### ISSUE THREAD

**Issue by antjkennedy** [senecajs/seneca-web#52]
Hi All,

I have just encountered a bug when using `express 4.13.4` with `seneca 1.2.0`. Using the quick example found in the [README.md](https://github.com/senecajs/seneca-web/blob/master/README.md) seneca-web will handle the first request, and then ignore subsequent attempts to access the defined routes.

Once in this state it looks like seneca is no longer called from the express router. Will update as/when I find out more.

### log

```
  express:application set "x-powered-by" to true +0ms
  express:application set "etag" to 'weak' +3ms
  express:application set "etag fn" to [Function: wetag] +0ms
  express:application set "env" to 'development' +0ms
  express:application set "query parser" to 'extended' +0ms
  express:application set "query parser fn" to [Function: parseExtendedQueryString] +0ms
  express:application set "subdomain offset" to 2 +1ms
  express:application set "trust proxy" to false +0ms
  express:application set "trust proxy fn" to [Function: trustNone] +0ms
  express:application booting in development mode +0ms
  express:application set "view" to [Function: View] +0ms
  express:application set "views" to '/home/vagrant/app/broken/views' +0ms
  express:application set "jsonp callback name" to 'callback' +0ms
  express:router use / query +1ms
  express:router:layer new / +0ms
  express:router use / expressInit +0ms
  express:router:layer new / +0ms
  express:router use / <anonymous> +1ms
  express:router:layer new / +0ms
2016-02-18T11:21:59.284Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	IN	e1l39rkpv8s4/lgkoqr7xcyes	name:basic,plugin:define,role:seneca,seq:1,tag:undefined	{role:seneca,plugin:define,name:basic,tag:null,seq:1}	ENTRY	(3xj7c)	-	-	GATE
2016-02-18T11:21:59.295Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	-       	OUT	qe5fg1psxqom/lgkoqr7xcyes		{}	EXIT		-	-	1	-
2016-02-18T11:21:59.296Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	OUT	e1l39rkpv8s4/lgkoqr7xcyes	name:basic,plugin:define,role:seneca,seq:1,tag:undefined	null	EXIT	(3xj7c)	-	-	510	GATE
2016-02-18T11:21:59.297Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	IN	us28zktq7xty/7ws14u9x8cia	name:seneca-jeptg0,plugin:define,role:seneca,seq:2,tag:undefined	{role:seneca,plugin:define,name:seneca-jeptg0,tag:null,seq:2}	ENTRY	(7ogju)	-	-	GATE
2016-02-18T11:21:59.298Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	-       	OUT	lyoydgkqsc40/7ws14u9x8cia		{}	EXIT		-	-	0	-
2016-02-18T11:21:59.298Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	OUT	us28zktq7xty/7ws14u9x8cia	name:seneca-jeptg0,plugin:define,role:seneca,seq:2,tag:undefined	null	EXIT	(7ogju)	-	-	473	GATE
2016-02-18T11:21:59.299Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	IN	j8xdsspawo6i/g9k63n5rl84y	name:seneca-s1vpnz,plugin:define,role:seneca,seq:3,tag:undefined	{role:seneca,plugin:define,name:seneca-s1vpnz,tag:null,seq:3}	ENTRY	(patbl)	-	-	GATE
2016-02-18T11:21:59.303Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	mem-store/1	IN	c2arqfdqnan5/g9k63n5rl84y	init:mem-store	{init:mem-store,tag:1}	ENTRY	(67wpr)	-	-	-
2016-02-18T11:21:59.304Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	mem-store/1	OUT	c2arqfdqnan5/g9k63n5rl84y	init:mem-store	null	EXIT	(67wpr)	-	-	1	-
2016-02-18T11:21:59.304Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	OUT	j8xdsspawo6i/g9k63n5rl84y	name:seneca-s1vpnz,plugin:define,role:seneca,seq:3,tag:undefined	null	EXIT	(patbl)	-	-	450	GATE
2016-02-18T11:21:59.304Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	IN	nubnx5j0hscm/kc3yitqi9v0r	name:seneca-ln545z,plugin:define,role:seneca,seq:4,tag:undefined	{role:seneca,plugin:define,name:seneca-ln545z,tag:null,seq:4}	ENTRY	(j3n1j)	-	-	GATE
2016-02-18T11:21:59.314Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	-       	OUT	317z3lpaeysx/kc3yitqi9v0r		{}	EXIT		-	-	0	-
2016-02-18T11:21:59.315Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	OUT	nubnx5j0hscm/kc3yitqi9v0r	name:seneca-ln545z,plugin:define,role:seneca,seq:4,tag:undefined	null	EXIT	(j3n1j)	-	-	301	GATE
2016-02-18T11:21:59.315Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	IN	bupgk7xmt53f/wescvsw9g7ya	name:seneca-77ny1f,plugin:define,role:seneca,seq:5,tag:undefined	{role:seneca,plugin:define,name:seneca-77ny1f,tag:null,seq:5}	ENTRY	(18vrt)	-	-	GATE
2016-02-18T11:21:59.338Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	web       	IN	i1epi29yw946/wescvsw9g7ya	init:web	{init:web,tag:null}	ENTRY	(d8640)	-	-	-
2016-02-18T11:21:59.338Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	web       	OUT	i1epi29yw946/wescvsw9g7ya	init:web	null	EXIT	(d8640)	-	-	1	-
2016-02-18T11:21:59.338Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	OUT	bupgk7xmt53f/wescvsw9g7ya	name:seneca-77ny1f,plugin:define,role:seneca,seq:5,tag:undefined	null	EXIT	(18vrt)	-	-	190	GATE
2016-02-18T11:21:59.339Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	web       	IN	3c6b1u01m1p8/wescvsw9g7ya	role:web	{role:web,plugin:web,config:{prefix:/seneca}}	ENTRY	(axg33)	-	-	-
2016-02-18T11:21:59.340Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	web       	OUT	3c6b1u01m1p8/wescvsw9g7ya	role:web	null	EXIT	(axg33)	-	-	2	-
2016-02-18T11:21:59.340Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	basic       	IN	7k2kiug8wu38/wescvsw9g7ya	cmd:push,note:true,role:basic	{role:basic,note:true,cmd:push,key:admin/units,value:{unit:web-service,spec:{title:Web Services,ng:{}},content:	ENTRY	(3g0nx)	-	-	-
2016-02-18T11:21:59.340Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	basic       	OUT	7k2kiug8wu38/wescvsw9g7ya	cmd:push,note:true,role:basic	null	EXIT	(3g0nx)	-	-	2	-
2016-02-18T11:21:59.341Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	web       	IN	wgdvfva4y8h2/jypxfbnghncj	role:web	{use:{prefix:/my-api,pin:{role:api,cmd:*},map:{zig:true,bar:{},qaz:{}}},role:web}	ENTRY	(axg33)	-	-	-
2016-02-18T11:21:59.341Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	web       	OUT	wgdvfva4y8h2/jypxfbnghncj	role:web	null	EXIT	(axg33)	-	-	189	-
  express:router dispatching GET /my-api/bar?zoo=5 +11s
  express:router query  : /my-api/bar?zoo=5 +1ms
  express:router expressInit  : /my-api/bar?zoo=5 +2ms
  express:router <anonymous>  : /my-api/bar?zoo=5 +0ms
2016-02-18T11:22:10.624Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	web       	OUT	wgdvfva4y8h2/jypxfbnghncj	role:web	null	EXIT	(axg33)	-	-	11469	-
2016-02-18T11:22:10.628Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	IN	o0z96f965ui6/rz8hni9fnnjq	cmd:bar,role:api	{cmd:bar,role:api,zoo:5,did:(7hx9w),fixedargs:{},context:{}}	ENTRY	(vlry7)	-	-	-
2016-02-18T11:22:10.629Z 6r2lg1duvl0j/1455794518666/4871/undefined DEBUG	act	root$       	OUT	o0z96f965ui6/rz8hni9fnnjq	cmd:bar,role:api	{bar:5b}	EXIT	(vlry7)	-	-	2	-
  express:router dispatching GET /my-api/bar?zoo=5 +2s
  express:router query  : /my-api/bar?zoo=5 +0ms
  express:router expressInit  : /my-api/bar?zoo=5 +1ms
  express:router <anonymous>  : /my-api/bar?zoo=5 +0ms
```

### example.js

```javascript
var seneca = require('seneca')()


seneca.add('role:api,cmd:zig',function(args,done){
  done(null,{bar:args.zoo+'g'})
})

seneca.add('role:api,cmd:bar',function(args,done){
  done(null,{bar:args.zoo+'b'})
})

seneca.add('role:api,cmd:qaz',function(args,done){
  done(null,{qaz:args.zoo+'z'})
})


seneca.act('role:web',{use:{
  
  // define some routes that start with /my-api
  prefix: '/my-api',

  // use action patterns where role has the value 'api' and cmd has some defined value
  pin: {role:'api',cmd:'*'},

  // for each value of cmd, match some HTTP method, and use the
  // query parameters as values for the action
  map:{
    zig: true,                // GET is the default
    bar: {GET:true},          // explicitly accepting GETs
    qaz: {GET:true,POST:true} // accepting both GETs and POSTs
  }
}})

var express = require('express')
var app = express()


// This is how you integrate Seneca with Express
app.use( seneca.export('web') )

app.listen(3000)
```

### package.json

```json
{
  "name": "express-seneca-web",
  "version": "0.0.0",
  "description": "example of broken seneca/express routing",
  "main": "main.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "express": "^4.13.4",
    "seneca": "^1.2.0"
  }
}
```

**Comment by mcdonnelldean** [comment]
Adding note, I've been using seneca 1.2.0 all morning and no issue so this is in seneca-web somewhere I bet.

**Comment by antjkennedy** [comment]
@mcdonnelldean almost definitely, the good news however is that is fixed in the latest commit.

After updating my package.json to

```
{
  "dependencies" : {
    "express": "^4.13.4",
    "seneca": "^1.1.0",
    "seneca-web": "git+https://github.com/senecajs/seneca-web.git#5b6dcd874395f574dc8536d3b0b452b27ac525c6"
  }
}
```

and changing the seneca import to

```
var seneca = require('seneca')({
  default_plugins:{
    web:false
  }
});
seneca.use(require('seneca-web'));
```

it all appears to be working as expected now although importing `seneca-web 0.7` is broken.

**Comment by geek** [comment]
@antjkennedy seneca-web 0.7.1 is now published with that fix... feel free to point to that in your package

**Comment by nfantone** [comment]
Same over here. Arrived at the same solution. Should we expect a release for `seneca` that includes 0.7.1 any time soon?

**Comment by mcdonnelldean** [comment]
This is resolved with #79
